### PR TITLE
Also define function EclipseState::hasInputLgr

### LINKED ISSUE #3787: [BugReport] EclipseState::hasInputLgrs() definition missing
Function  EclipseState::hasInputLGR() is declared (EclipseState.hpp:102), however, it is not defined (in EclipseState.cpp).

### PR DESCRIPTION
**Pull request by blattms**
Otherwise people will experience linker error when trying to use it.

Closes #3787

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

**Comment by blattms** [comment]
:flushed: fixed some locally without amending the commit and pushing. How hard can it be :see_no_evil:

**Comment by blattms** [comment]
jenkins build this please

**Comment by aritorto** [comment]
It was just a "convention typo" :) (lowercase vs capital letters)

File: src/opm/input/eclipse/EclipseState/EclipseState.cpp [hunk 1]
```diff
@@ -275,6 +275,10 @@ namespace Opm {
         return !m_inputNnc.input().empty();
     }
 
+    bool EclipseState::hasInputLGR() const {
+        return m_lgrs.size();
+    }
```
Comment (bska) [inline src/opm/input/eclipse/EclipseState/EclipseState.cpp:280]: Could you write this as `size() != 0`, please?  Ideally, there'd be an `empty()` predicate on the `OrderedMap<>` and we could write this as
```C++
return !m_lgrs.empty()
```
instead, but `size() != 0` is the next best thing.

File: tests/parser/LgrTests.cpp [hunk 2]
```diff
@@ -101,6 +101,7 @@ SCHEDULE
     Opm::EclipseState state(deck);
     Opm::LgrCollection lgrs = state.getLgrs();
 
+    BOOST_CHECK(state.hasInputLGR());
```
Comment (bska) [inline tests/parser/LgrTests.cpp:104]: Would you mind using `BOOST_CHECK_MESSAGE()` for this?  A readable diagnostic message with more context is usually preferable to just "Test X failed".

**Review by bska — COMMENTED** [review]
Thank you for plugging this hole. Just a couple of comments regarding the implementation details.

**Comment by blattms** [comment]
Comments addressed.

**Comment by blattms** [comment]
jenkins build this please

**Review by bska — APPROVED** [review]
Thanks a lot for the updates.  I'll merge into master.

**Pull request merged**